applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -488,247 +602,103 @@ int handle_at_gps(enum at_cmd_type cmd_type)
-			if (err == 0) {
-				int64_t gps_sec = utc_to_gps_sec(utc_now_ms/1000);
-				struct nrf_modem_gnss_agps_data_system_time_and_sv_tow gps_time
-					= { 0 };
-
-				gps_sec_to_day_time(gps_sec, &gps_time.date_day,
-						    &gps_time.time_full_s);
-				err = nrf_modem_gnss_agps_write(&gps_time, sizeof(gps_time),
-						NRF_MODEM_GNSS_AGPS_GPS_SYSTEM_CLOCK_AND_TOWS);
-				if (err) {
-					LOG_WRN("Fail to inject time, error %d", err);
+			if (interval == 1) {
+				/* Continuous navigation mode; the timeout must not be given. */
+				if (param_count != TIMEOUT_IDX) {
+					return -EINVAL;
+				}
+			} else {
+				if (param_count == TIMEOUT_IDX) {
+					timeout = 60;  /* default value */
 				} else {
-					LOG_DBG("Inject time (GPS day %u, GPS time of day %u)",
-						gps_time.date_day, gps_time.time_full_s);
+					err = at_params_unsigned_short_get(
+						&slm_at_param_list, TIMEOUT_IDX, &timeout);
+					if (err || param_count != MAX_PARAM_COUNT) {
+						return -EINVAL;
+					}
 				}
-			}
-
-			err = gnss_startup(RUN_TYPE_AGPS);
-		} else if (op == AGPS_STOP && run_type == RUN_TYPE_AGPS) {
-			err = gnss_shutdown();
-		} else {
-			err = -EINVAL;
-		} break;
-
-	case AT_CMD_TYPE_READ_COMMAND:
-		rsp_send("\r\n#XAGPS: %d,%d\r\n", (int)is_gnss_activated(), run_status);
-		err = 0;
-		break;
-
-	case AT_CMD_TYPE_TEST_COMMAND:
-		rsp_send("\r\n#XAGPS: (%d,%d),<interval>,<timeout>\r\n", AGPS_STOP, AGPS_START);
-		err = 0;
-		break;
-
-	default:
-		break;
-	}
-
-	return err;
-}
-#endif /* CONFIG_NRF_CLOUD_AGPS */
-
-#if defined(CONFIG_SLM_NRF_CLOUD) && defined(CONFIG_NRF_CLOUD_PGPS)
-/* Handles AT#XPGPS commands. */
-int handle_at_pgps(enum at_cmd_type cmd_type)
-{
-	int err = -EINVAL;
-	uint16_t op;
-	uint16_t interval;
-	uint16_t timeout;
-
-	switch (cmd_type) {
-	case AT_CMD_TYPE_SET_COMMAND:
-		err = at_params_unsigned_short_get(&slm_at_param_list, 1, &op);
-		if (err < 0) {
-			return err;
-		}
-		if (op == PGPS_START && slm_nrf_cloud_ready && run_type == RUN_TYPE_NONE) {
-			err = at_params_unsigned_short_get(&slm_at_param_list, 2, &interval);
-			if (err < 0) {
-				return err;
-			}
-			/* GNSS API spec check, P-GPS is used in periodic mode only */
-			if (interval < 10 || interval > 65535) {
-				return -EINVAL;
-			}
-			/* no scheduled downloads for peoridical tracking */
-			err = nrf_modem_gnss_use_case_set(
-					NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START |
-					NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE);
-			if (err) {
-				LOG_ERR("Failed to set use case, error: %d", err);
-				return err;
-			}
-			err = nrf_modem_gnss_fix_interval_set(interval);
-			if (err) {
-				LOG_ERR("Failed to set fix interval, error: %d", err);
-				return err;
-			}
-			if (at_params_unsigned_short_get(&slm_at_param_list, 3, &timeout) == 0) {
+				/* Make sure to configure the timeout in all cases.
+				 * An old value might otherwise remain from a previous GPS run.
+				 */
 				err = nrf_modem_gnss_fix_retry_set(timeout);
```

Comment:
Above comment seems to bit misleading as this is not set if the timeout is not given. If it possible to have this from previous GPS run, then this might be problematic in case we first run with timeout and after that run continous.